pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -857,14 +857,20 @@ func (og *operationGenerator) resizeFileSystem(volumeToMount VolumeToMount, rsOp
 	return true, nil
 }
 
-func (og *operationGenerator) getKubeClient(tenant string) clientset.Interface {
+func (og *operationGenerator) getKubeClient(boundPodUID types.UID) clientset.Interface {
 	if len(og.kubeTenantPartitionClients) == 1 {
 		return og.kubeTenantPartitionClients[0]
 	} else {
-		return kubeclientmanager.ClientManager.GetTPClient(og.kubeTenantPartitionClients, tenant)
+		return kubeclientmanager.ClientManager.GetTPClient(og.kubeTenantPartitionClients, boundPodUID)
 	}
 }
 
+// supposed to be called by controller (e.g. expandController) inside of KCM,
+// assuming the controller is within TP env in scale-out cluster.
+func (og *operationGenerator) getDefaultKubeClient() clientset.Interface {
+	return og.kubeTenantPartitionClients[0]
```

Comment:
Is operationGenerator used both in TP and RP? If not, it should not have multiple TPClients in TP controllers.

Reply:
it is used both in kubelet(which is at RP) and KCM controller (which is supposedly at TP)